File: backend/src/ApplicationCore/Entities/Orders/Address.cs
using PartyKlinest.ApplicationCore.Interfaces;

namespace PartyKlinest.ApplicationCore.Entities.Orders
{
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
    public record Address : IAggregateRoot
    {
        public long AddressId { get; set; }
        public string Street { get; set; }
        public string BuildingNumber { get; set; }
        public int? FlatNumber { get; set; }
        public string City { get; set; }
        public string PostalCode { get; set; }
        public string Country { get; set; }
    }
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
}

File: frontend/src/DataClasses/Address.ts
interface Address {
    street: string;
    buildingNo: string;
    flatNo?: number;
    city: string;
    postalCode: string;
    country: string;
}

export function isAddressCorrect(address: Address): boolean {
    return address.street !== '' &&
        address.buildingNo !== '' &&
        address.flatNo !== 0 && (address.flatNo === undefined || !isNaN(address.flatNo)) &&
        address.city !== '' &&
        address.postalCode !== '' &&
        address.country !== '';
}

export function emptyAddress(): Address {
    return {
        street: '',
        buildingNo: '',
        flatNo: undefined,
        city: '',
        postalCode: '',
        country: ''
    };
}

export default Address;
